pkg/cpuinfo/cpuinfo.go

```diff
@@ -453,3 +517,26 @@ func combinePath(value string, combineWith ...string) string {
 		return filepath.Join(all...)
 	}
 }
+func (t *CPUTopology) CPUsPerCore() int {
+	if t.NumCores == 0 {
+		return 0
+	}
+	return t.NumCPUs / t.NumCores
+}
+
+func (t *CPUTopology) CPUsPerSocket() int {
+	if t.NumSockets == 0 {
+		return 0
+	}
+	return t.NumCPUs / t.NumSockets
+}
+
+func (t *CPUTopology) CPUsPerUncore() int {
+	if t.NumUncoreCache == 0 {
+		// Avoid division by zero. If there are no uncore caches, then
+		// no CPUs can be in one.
+		return 0
+	}
+	// Note: this is an approximation that assumes all uncore caches have the same number of CPUs.
+	return t.NumCPUs / t.NumUncoreCache
+}
```

Comment:
I'm really torn here. On one hand we should not block here, but this is a really egregious oversimplification has been a subtle source of troubles in cpumanager. The easiest case to consider is a user offlining a set of CPUs asymmetrically, e.g. only on one NUMA node, or LLC complex. All the math goes haywire.

This is probably one of the best areas on which we should diverge from cpumanager code (which is pretty much unfixable at this point in time) and do better. I'm confident that this "do better" doesn't require a totalrewrite or a super complex representation.

I think the best way forward is be very aware of this fragility and plan improvements for the next milestones.